tests/pytorch_tests/pruning_tests/test_pretrained_models.py

```diff
@@ -0,0 +1,147 @@
+            dense_model (Model): The pre-trained Pytorch model to be pruned.
+            test_retraining (bool): If True, retrain the pruned model on dummy data to test stability.
+
+        This function calculates the number of parameters in the dense model, performs pruning to achieve
+        the desired compression rate, and validates the actual compression rate achieved. It also tests
+        if the outputs of the pruned model are similar to the dense model, and ensures that pruned layers
+        respect the importance scores. If `test_retraining` is True, it further validates the model's
+        performance after retraining.
+        """
+        # Calculate the number of parameters in the dense model.
+        dense_nparams = count_model_prunable_params(dense_model)
+
+        # Perform pruning on the dense model.
+        pruned_model, pruning_info = mct.pruning.pytorch_pruning_experimental(
+            model=dense_model,
+            target_kpi=mct.KPI(weights_memory=dense_nparams * FP32_BYTES_PER_PARAMETER * cr),
+            representative_data_gen=self.representative_dataset,
+            pruning_config=mct.pruning.PruningConfig(
+                num_score_approximations=1,
+                importance_metric=TestImportanceMetric.RANDOM)
+        )
+
+        # Calculate the actual compression rate achieved after pruning.
+        pruned_nparams = count_model_prunable_params(pruned_model)
+        actual_cr = pruned_nparams / dense_nparams
+        print(f"Target remaining cr: {cr * 100}, Actual remaining cr: {actual_cr * 100}")
+
+        input_tensor = next(self.representative_dataset())[0]
+        pruned_outputs = pruned_model(to_torch_tensor(input_tensor))
+
+        # Ensure pruned layers had lower importance scores than the channels
+        # that remained.
+        for layer_name, layer_mask in pruning_info.pruning_masks.items():
+            if 0 in layer_mask:
+                layer_scores = pruning_info.importance_scores[layer_name]
+                min_score_remained = min(layer_scores[layer_mask.astype("bool")])
+                max_score_removed = max(layer_scores[(1 - layer_mask).astype("bool")])
+                self.assertTrue(max_score_removed <= min_score_remained,
+                                f"Expected remaining channels to have higher scores"
+                                f"than pruned channels but found remained channel with score"
+                                f"{min_score_remained} and found pruned channel with"
+                                f"score {max_score_removed}")
+
+        # Validate that the actual compression rate does not exceed the target compression rate.
+        self.assertTrue(actual_cr <= cr,
+                        f"Expected the actual compression rate: {actual_cr} to not exceed the target compression "
+                        f"rate: {cr}")
+
+
+# Function to generate an infinite stream of dummy images and labels
+def dummy_data_generator():
```

Comment:
If you're not using it, then remove it. But I think it's better to add a flag that indicates whether to test it or not, which is disabled by default (and at least test it once when you run it locally to see if the expected post-pruning retraining is possible).

Reply:
I added retraining